OpenRA.Mods.Common/Activities/Repair.cs

```diff
@@ -33,12 +33,15 @@ public Repair(Actor self, Actor host, WDist closeEnough)
 		{
 			this.host = Target.FromActor(host);
 			this.closeEnough = closeEnough;
-			repairsUnits = host.Info.TraitInfo<RepairsUnitsInfo>();
+			repairsUnits = host.Trait<RepairsUnits>();
 			health = self.TraitOrDefault<Health>();
 		}
 
 		public override Activity Tick(Actor self)
 		{
+			if (repairsUnits.IsTraitDisabled)
```

Comment:
There's a couple of issues here:
* This jams up the unit completely, preventing you from ordering it off the repair pad.
* We have inconsistent behaviour for units that were on vs off the pad when it became disabled: things on the pad wait/are stuck, while things off the pad can't enter it.

The problem is that there are two different types of disabling that we care about here:
1. Adding/removing the ability to repair (e.g. via a building upgrade).  This should cancel any active repair when disabled.
2. Temporarily pausing (e.g. emp, manual power down).  This should pause the active repair, and new units should be allowed to dock (but repairs won't start until it is reenabled).

The solution should be straightforward, adding a new `ActiveCondition` field to `RepairsUnits` that implements (2), and tweak (1) to `return NextActivity`.